fix: group-music-profile, member-music-profile 수정

ownue opened the pull request:
## 📌 관련 이슈
#11

## ✨ 작업 내용
- NOT NULL이었던 속성들을 배치 전 조회를 고려해 NULL로 변경
- status 값에 따라 NULL 허용 여부를 조절하도록 수정
- ProfileStatus enum 분리 (status 값 추가 및 변경 -> NOT_AVAILABLE, PROCESSING, READY로 최종)
- createdDate와 LastModifiedDate 어노테이션 추가
- ErrorCode로 무결성 관리
- status의 null 값 가드

## 📸 스크린샷(선택)
X

## 📚 레퍼런스 (또는 새로 알게 된 내용) 혹은 궁금한 사항들
X

ownue committed:
fix: group-music-profile, member-music-profile 수정

- NOT NULL이었던 속성들을 배치 전 조회를 고려해 NULL로 변경
- status 값에 따라 NULL 허용 여부를 조절하도록 수정

hyomee2 commented on src/main/java/com/example/muaring/domain/group/entity/GroupMusicProfile.java:
```diff
@@ -20,63 +25,144 @@ public class GroupMusicProfile {
     private LocalDateTime updatedAt;
 
     // status enum으로 관리, 필요할 시 파일 분리할 것
     public enum ProfileStatus {
```
혹시 여기서 ProfileStatus가 어떤거에 대한 컬럼이죠..??

ownue replied:
컬럼 아니구 status에 사용하는 enum입니다!!
status는 언제 값이 변경되냐면
1. 처음 그룹이 생성됐을 경우, ```NOT_AVAILABLE``` -> 그룹 음악 특성 데이터 생성, 아직 업로드 된 곡이 없으므로 상세 특성은 미존재, 관련 데이터 조회 불가
2. 매일 새벽 배치 중인 경우, ```PROCESSING``` -> 관련 데이터 조회 불가
3. 그 외 경우, ```READY``` -> 관련 데이터 조회 가능

이렇게입니다!!

hyomee2 commented on src/main/java/com/example/muaring/domain/group/entity/GroupMusicProfile.java:
```diff
@@ -20,63 +25,144 @@ public class GroupMusicProfile {
     private LocalDateTime createdAt;
 
-    @Column(name = "updated_at")
+    @Column(name = "updated_at", nullable = false)
```
@LastModifiedDate 사용하면 좋을 것 같습니다!

hyomee2 commented on src/main/java/com/example/muaring/domain/member/entity/MemberMusicProfile.java:
```diff
@@ -68,6 +70,67 @@ public class MemberMusicProfile {
 
     // 일단 한 파일에 넣었는데 분리 필요하면 분리하기
     public enum ProfileStatus {
```
이 enum이 MemberMusicProfile이랑 GroupMusicProfile에서 공통으로 사용되는거 같은데 따로 빼는건 어떤가요??

ownue replied:
넵!

ownue committed:
chore: ProfileStatus enum 분리 및 어노테이션 추가 등

- PR 코드 리뷰 맞춰 일부 수정
- 공통으로 쓰이는 ProfileStatus 별도 파일로 분리
- createdDate와 LastModifiedDate 어노테이션 추가
- ErrorCode 추가하여 무결성 관리

ownue committed:
chore: entity 내 팩토리 메서드 수정 및 status null 방지

- 어노테이션 추가로 인해 NOW()를 삭제
- status에 null 값이 들어올 경우, NOT_AVAILABLE로 기본값 설정

romdyfo reviewed (approved):
확인입니다! 그룹 엔티티 생성 감사해요~!!

hyomee2 reviewed (approved):
굿입니다!
